### DISCUSSION TITLE
i have solved the classic problems. Because it is France original but we use it in NL/BE/GE/IT/etc.

### DISCUSSION THREAD

**Discussion by ghost** [PrestaShop/PrestaShop#31067] [Ideas]
### The problem

i have solved it but know where to place. I know you  and rest don't give a. but if there are so many people asking sins 2012 about same problems in 10 lang. You know they can use it. Maybe even you can sell more. 

### The solution

public function getTemplateVarShop()
    {
        $address = $this->context->shop->getAddress();

        $urls = $this->getTemplateVarUrls();
        $psImageUrl = $urls['img_ps_url'] ?? _PS_IMG_;

        $shop = [
            'id' => $this->context->shop->id,
            'name' => Configuration::get('PS_SHOP_NAME'),
            'email' => Configuration::get('PS_SHOP_EMAIL'),
            'registration_number' => Configuration::get('PS_SHOP_SIRET'),
			'registration_number' => Configuration::get('PS_SHOP_vat_number'),
	        'registration_number' => Configuration::get('PS_SHOP_IBAN'),

            'long' => Configuration::get('PS_STORES_CENTER_LONG'),
            'lat' => Configuration::get('PS_STORES_CENTER_LAT'),

            'logo' => Configuration::hasKey('PS_LOGO') ? $psImageUrl . Configuration::get('PS_LOGO') : '',
            'logo_details' => $this->getShopLogo(),
            'stores_icon' => Configuration::hasKey('PS_STORES_ICON') ? $psImageUrl . Configuration::get('PS_STORES_ICON') : '',
            'favicon' => Configuration::hasKey('PS_FAVICON') ? $psImageUrl . Configuration::get('PS_FAVICON') : '',
            'favicon_update_time' => Configuration::get('PS_IMG_UPDATE_TIME'),

            'address' => [
                'formatted' => AddressFormat::generateAddress($address, [], '<br>'),
                'address1' => $address->address1,
                'address2' => $address->address2,
                'postcode' => $address->postcode,
                'city' => $address->city,
                'state' => (new State($address->id_state))->name,
                'country' => (new Country($address->id_country))->name[$this->context->language->id],
            ],
            'phone' => Configuration::get('PS_SHOP_PHONE'),
            'fax' => Configuration::get('PS_SHOP_FAX'),
        ];

        return $shop;
    }

### Alternatives

more code is change. But it all works.

   /**
     * Validate SIRET Code.
     *
     * @param string $siret SIRET Code
     *
     * @return bool Return true if is valid
     */
 public static function isSiret($siret)
    {
    return (bool)preg_match('/^[0-9]{8,11}$/', $siret);
    }

    /**
     * Validate APE Code/ VAT code.
     *
     * @param string $vat APE Code VAT code
     *
     * @return bool Return true if is valid
     */
    public static function isVat($vat)
    {
        return (bool) preg_match('/^[0-9]{8,17}[a-zA-Z]{3}$/S', $vat);
    }
	    /**
     * Validate APE Code/ VAT code.
     *
     * @param string $vat APE Code VAT code
     *
     * @return bool Return true if is valid
     */
    public static function isvat_number($vat_number)
    {
        return (bool) preg_match('/^\d[0-9]{8,17}[a-zA-Z]{3}$/', $vat_number);
    }
	$vat_number = 'NL001947572B19';

// Create a new SOAP client
$client = new SoapClient("http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl");

// Send the VAT number to the VIES web service
$result = $client->checkVat(array('countryCode' => 'NL', 'vatNumber' => $vat_number));

// Check the result
if ($result->valid) {
    echo "The VAT number is valid";
} else {
    echo "The VAT number is invalid";
}

	    /**
     * Validate Iban code.
     *
     * @param string $Iban code
     *
     * @return bool Return true if is valid
     */
    public static function isIban($iban)
    {
        return (bool) preg_match('/^[0-9]{14,18}[a-zA-Z]{6}$/S', $iban);
    }

### Additional context

And i have 0,0 diploma or programming expert . So hard was it not. If you look up SIRET and APE and VAT and vat_number you can get verry far. Just edit extra IBAN (bank information) and done. You got a good working back and and front. 

 

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by ghost** [comment]
{extends file='page.tpl'}

{block name='page_title'}
  {l s='Our stores' d='Shop.Theme.Global'}
{/block}

{block name='page_content_container'}
  <section id="content" class="page-content page-stores">

    {foreach $stores as $store}
      <article id="store-{$store.id}" class="store-item card">
        <div class="store-item-container clearfix">
          <div class="col-md-3 store-picture hidden-sm-down">
            <img
              src="{$store.image.bySize.stores_default.url}"
              {if !empty($store.image.legend)}
                alt="{$store.image.legend}"
                title="{$store.image.legend}"
              {else}
                alt="{$store.name}"
              {/if}
            >
          </div>
          <div class="col-md-5 col-sm-7 col-xs-12 store-description">
            <p class="h3 card-title">{$store.name}</p>
            <address>{$store.address.formatted nofilter}</address>
            {if $store.note || $store.phone || $store.fax || $store.siret || $store.vat_number || $store.iban || $store.email}
              <a data-toggle="collapse" href="#about-{$store.id}" aria-expanded="false" aria-controls="about-{$store.id}"><strong>{l s='About and Contact' d='Shop.Theme.Global'}</strong><i class="material-icons">&#xE409;</i></a>
            {/if}
          </div>
          <div class="col-md-4 col-sm-5 col-xs-12 divide-left">
            <table>
              {foreach $store.business_hours as $day}
              <tr>
                <th>{$day.day|truncate:4:'.'}</th>
                <td>
                  <ul>
                  {foreach $day.hours as $h}
                    <li>{$h}</li>
                  {/foreach}
                  </ul>
                </td>
              </tr>
              {/foreach}
            </table>
          </div>
        </div>
        <footer id="about-{$store.id}" class="collapse">
          <div class="store-item-footer divide-top">
            {if $store.note}
              <div class="card-block">
                <p class="text-justify">{$store.note}</p>
              </div>
            {/if}
            <ul class="card-block">
              {if $store.phone}
                <li><i class="material-icons">&#xE0B0;</i>{$store.phone}</li>
              {/if}
              {if $store.fax}
                <li><i class="material-icons">&#xE8AD;</i>{$store.fax}</li>
              {/if}
			  {if $store.siret}
                <li><i class="material-icons">&#xE88F;</i>{$store.siret}</li>
              {/if}
              {if $store.vat_number}
                <li><i class="material-icons">&#xE88F;</i>{$store.vat_number}</li>
              {/if}
              {if $store.iban}
                <li><i class="material-icons">&#xE88F;</i>{$store.iban}</li>
              {/if}
              {if $store.email}
                <li><i class="material-icons">&#xE0BE;</i>{$store.email}</li>
              {/if}
            </ul>
          </div>
        </footer>
      </article>
    {/foreach}

  </section>
{/block}

**Comment by ghost** [comment]
<div class="block-contact col-md-3 links wrapper">
  <div class="title clearfix hidden-md-up" data-target="#contact-infos" data-toggle="collapse">
    <span class="h3">{l s='Store information' d='Shop.Theme.Global'}</span>
    <span class="float-xs-right">
      <span class="navbar-toggler collapse-icons">
        <i class="material-icons add">keyboard_arrow_down</i>
        <i class="material-icons remove">keyboard_arrow_up</i>
      </span>
    </span>
  </div>

  <p class="h4 text-uppercase block-contact-title hidden-sm-down">{l s='Store information' d='Shop.Theme.Global'}</p>
  <div id="contact-infos" class="collapse">
    {$contact_infos.address.formatted nofilter}
    {if $contact_infos.phone}
      <br>
      {* [1][/1] is for a HTML tag. *}
      {l s='Call us: [1]%phone%[/1]'
        sprintf=[
        '[1]' => '<span><strong>',
        '[/1]' => '</strong></span>',
        '%phone%' => $contact_infos.phone
        ]
        d='Shop.Theme.Global'
      }
    {/if}
    {if $contact_infos.fax}
      <br>
      {* [1][/1] is for a HTML tag. *}
      {l
        s='Fax: [1]%fax%[/1]'
        sprintf=[
        '[1]' => '<span><strong>',
        '[/1]' => '</strong></span>',
          '%fax%' => $contact_infos.fax
        ]
        d='Shop.Theme.Global'
      }
    {/if}
	{if $contact_infos.siret}
      <br>
      {* [1][/1] is for a HTML tag. *}
      {l s='Siret: [1]%siret%[/1]'
        sprintf=[
        '[1]' => '<span><strong>',
        '[/1]' => '</strong></span>',
        '%siret%' => $contact_infos.siret
        ]
        d='Shop.Theme.Global'
      }
    {/if}
    {if $contact_infos.iban}
      <br>
      {* [1][/1] is for a HTML tag. *}
      {l s='Iban: [1]%iban%[/1]'
        sprintf=[
        '[1]' => '<span><strong>',
        '[/1]' => '</strong></span>',
        '%iban%' => $contact_infos.iban
        ]
        d='Shop.Theme.Global'
      }
    {/if}
{if $contact_infos.vat_number}
      <br>
      {* [1][/1] is for a HTML tag. *}
      {l s='VAT number: [1]%vat_number%[/1]'
        sprintf=[
        '[1]' => '<span><strong>',
        '[/1]' => '</strong></span>',
        '%vat_number%' => $contact_infos.vat_number
        ]
        d='Shop.Theme.Global'
      }
{/if}
    {if $contact_infos.email && $display_email}
      <br>
        {l s='Email us:' d='Shop.Theme.Global'}
        {mailto address=$contact_infos.email encode="javascript"}
    {/if}
  </div>
</div>

**Comment by ghost** [comment]
<pre>
  <code>
&lt;div class="contact-rich"&gt;
  &lt;h4&gt;{l s='Store information' d='Shop.Theme.Global'}&lt;/h4&gt;
  &lt;div class="block"&gt;
    &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&amp;#xE55F;&lt;/i&gt;&lt;/div&gt;
    &lt;div class="data"&gt;{$contact_infos.address.formatted nofilter}&lt;/div&gt;
  &lt;/div&gt;
  {if $contact_infos.phone}
    &lt;hr/&gt;
    &lt;div class="block"&gt;
      &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&amp;#xE0CD;&lt;/i&gt;&lt;/div&gt;
      &lt;div class="data"&gt;
        {l s='Call us:' d='Shop.Theme.Global'}&lt;br/&gt;
        &lt;a href="tel:{$contact_infos.phone}"&gt;{$contact_infos.phone}&lt;/a&gt;
       &lt;/div&gt;
    &lt;/div&gt;
  {/if}
  {if $contact_infos.fax}
    &lt;hr/&gt;
    &lt;div class="block"&gt;
      &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&amp;#xE0DF;&lt;/i&gt;&lt;/div&gt;
      &lt;div class="data"&gt;
        {l s='Fax:' d='Shop.Theme.Global'}&lt;br/&gt;
        {$contact_infos.fax}
      &lt;/div&gt;
    &lt;/div&gt;
  {/if}
 {if $contact_infos.siret}
    &lt;hr/&gt;
    &lt;div class="block"&gt;
      &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&amp;#xE88F;&lt;/i&gt;&lt;/div&gt;
      &lt;div class="data"&gt;
        {l s='Siret:' d='Shop.Theme.Global'}&lt;br/&gt;
        &lt;a href="info:{$contact_infos.siret}"&gt;{$contact_infos.siret}&lt;/a&gt;
       &lt;/div&gt;
    &lt;/div&gt;
  {/if}
    {if $contact_infos.vat_number}
&lt;hr/&gt;
&lt;div class="block"&gt;
  &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&#xE88F;&lt;/i&gt;&lt;/div&gt;
  &lt;div class="data"&gt;
    {l s='VAT number:' d='Shop.Theme.Global'}&lt;br/&gt;
    &lt;a href="info:{$contact_infos.vat_number}"&gt;{$contact_infos.vat_number}&lt;/a&gt;
   &lt;/div&gt;
&lt;/div&gt;
  {/if}
&lt;hr/&gt;
&lt;div class="block"&gt;
  &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&#xE88F;&lt;/i&gt;&lt;/div&gt;
  &lt;div class="data"&gt;
    {l s='Iban:' d='Shop.Theme.Global'}&lt;br/&gt;
    &lt;a href="info:{$contact_infos.iban}"&gt;{$contact_infos.iban}&lt;/a&gt;
   &lt;/div&gt;
&lt;/div&gt;
  {/if}
  {if $contact_infos.email && $display_email}
&lt;hr/&gt;
&lt;div class="block"&gt;
  &lt;div class="icon"&gt;&lt;i class="material-icons"&gt;&#xE158;&lt;/i&gt;&lt;/div&gt;
  &lt;div class="data email"&gt;
    {l s='Email us:' d='Shop.Theme.Global'}&lt;br/&gt;
  &lt;/div&gt;
  {mailto address=$contact_infos.email encode="javascript"}
&lt;/div&gt;
  {/if}
&lt;/div&gt;

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/116812745/215618947-fbf49a44-cdf9-4d72-b09c-da91c51dad93.png)

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/116812745/215619543-9e1765de-335d-44e3-9c7f-b77f44d33eb7.png)

**Comment by jf-viguier** [comment]
Can you explain more what do you suggest ?

**Reply by Gruttersshop** [reply]
i have uploaded new pickters

**Comment by kpodemski** [comment]
Hi @Gruttershop 

Please wait to send Pull Requests. Let's try to think about the best way to handle this. Please be patient :)

**Comment by Gruttersshop** [comment]
tank you kpodemski for blocking me. thought it was a opensource , but now i know it's who is longest can blok idears. But replaying on 
jf-viguier
Feb 3, 2023

Can you explain more what do you suggest ?

i had pictures. explaining that this prestashop is written for France. but also used in  EU. In Netherlands we need show the Siret, VAT and more. So i written the back-end and front. It shows correctly and can be edit in back and. But the project management won't see it or reconcile it. also in France to EU the APE and VAT are same. But are not. VAT is longer and APE sorter. Also prestashop even sees it as a paspoort ID???? So i corrected that problem to. But what i now? i'm just a stupid boy. 
![image](https://user-images.githubusercontent.com/124369064/216617967-bd20e11b-7a21-450e-88cb-65e5ed67dca3.png)
![image](https://user-images.githubusercontent.com/124369064/216618054-09ea6357-489a-43f9-bb31-cc550a9676e4.png)
![image](https://user-images.githubusercontent.com/124369064/216618154-4cf330ce-3309-4147-b742-1326dc044346.png)

**Comment by Gruttersshop** [comment]
isn't it better so? now more people can use the prestashop. and if you fill in 0. nothing it won't show. and it's now better for APE SIret vat_number DNI. More seprate. Not let's say APE is also VAT. APE=APE VAT=VAT SIRET=SIRET DNI=DNI 

don't mix them like a spoon can be like a fork to eat food. But a fork can't be like a spoon while eating soup. :)

**Comment by Gruttersshop** [comment]
class="material-icons">&#xE409;</i> for i

see that fax is know ??? uploading? that's your material-icon . I will look up and write the code. But don't think [kpodemski] can do it. He's working hard at blocking people.

**Comment by matks** [comment]
> tank you kpodemski for blocking me.

@kpodemski blocked you because you continue opening PR that do not match our validity criteria and you do not listen to what we tell you

Please check our Code of Conduct:
https://github.com/PrestaShop/PrestaShop/blob/0c2949b33dd3f57d49164d7988edc7926f0de13a/CODE_OF_CONDUCT.md
> Examples of behavior that contributes to creating a positive environment include:

> Using welcoming and inclusive language
Being respectful of differing viewpoints and experiences
Gracefully accepting constructive criticism
Focusing on what is best for the community
Showing empathy towards other community members

If you continue opening many Pull Requests that have no chance of being merged as they go against validity rules and continue complain with words like `He's working hard at blocking people.` I will take the necessary measures to make sure the mess stops.

Open source does not mean there is no rules.

**Reply by Gruttersshop** [reply]
if i mail you everyting you list? because if i forget a bo or fo you blok or delete.

**Reply by Gruttersshop** [reply]
do you even watch what it could look like?

**Comment by Gruttersshop** [comment]
Can i mail you the hole installation file's changes so you can put them integrate them for pull?? because it's a lot. It's front and BO ans tpl and more. not like 100. but started white rewriting 1 and after that it was 10.

**Comment by Gruttersshop** [comment]
i have installed prestashop with my changes now 13 times. It works. It place the changes and accepted them. Also BO and Front and DB. But know i have lot's of work but now ware to intergrate and if presta updates i will have to rewrite again. easer if it's intergrated. i'm also even translating it. I saw a bug that doesn't translate dutch but it does translate GE, EN,FR,IT, etc. by you but i don't want to report it there because it will blok me or won't fix the problem. 
![image](https://user-images.githubusercontent.com/124369064/216629789-fa421044-3d90-4fa7-b59e-7fd1de2d9550.png)


Producten

    Prices drop
    New products
    Best sales

Contact us
Sitemap
Stores  it's this. I looked it up and it's good in backend translated but doesn't see it in front end. outher lang it works. but not Duth.

**Comment by Gruttersshop** [comment]
![image](https://user-images.githubusercontent.com/124369064/216721600-e97d64dc-b29c-46c7-b41f-fa29188659d8.png)
![image](https://user-images.githubusercontent.com/124369064/216721631-4c718274-313d-46ef-bd54-76f58fdde240.png)
![image](https://user-images.githubusercontent.com/124369064/216721664-f11c8513-a722-47dd-96c4-302a96ad305f.png)
![image](https://user-images.githubusercontent.com/124369064/216721695-363e5548-1ba6-4446-a27c-a5db9e80d35e.png)
![image](https://user-images.githubusercontent.com/124369064/216721740-872e8f9c-3545-4002-ac57-3d06ed319fa6.png)

**Comment by Gruttersshop** [comment]
it's maybe strange that b2b has SIRET, VAT, APE and DNI for costumers. But shop not. I have also included it. But in Netherlands we don't use APE code or DNI. But it's a prestashop for EU? Or i'm i wrong? This is what i made. So if you like, pm me at webmaster@grutterswinkel.nl a will send the file to only the project-manager. Because i may not upload it or can't upload it because i don't usually make write in these program's langues. And i don't want to do you wrong.

**Comment by Gruttersshop** [comment]
installed with the rewritten installation file and FTP. and if you use demo. the languish is correct in Dutch?? So it's only wrong without demo.

**Comment by Gruttersshop** [comment]
CREATE TABLE `PREFIX_store` (
  `id_store` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `id_country` int(10) unsigned NOT NULL,
  `id_state` int(10) unsigned DEFAULT NULL,
  `name` varchar(128) NOT NULL,
  `address1` varchar(128) NOT NULL,
  `address2` varchar(128) DEFAULT NULL,
  `city` varchar(64) NOT NULL,
  `postcode` varchar(12) NOT NULL,
  `latitude` float(10,6) DEFAULT NULL,
  `longitude` float(10,6) DEFAULT NULL,
  `hours` varchar(254) DEFAULT NULL,
  `phone` varchar(16) DEFAULT NULL,
  `fax` varchar(16) DEFAULT NULL,
  `email` varchar(128) DEFAULT NULL,
  `siret` varchar(18) DEFAULT NULL,
  `vat_number` varchar(28) DEFAULT NULL,
  `iban` varchar(28) DEFAULT NULL,
  `APE` varchar(10) DEFAULT NULL,
  `dni` varchar(16) DEFAULT NULL,
  `note` text,

**Comment by Gruttersshop** [comment]
CREATE TABLE IF NOT EXISTS `PREFIX_store_lang` (
  `id_store` int(11) unsigned NOT NULL,
  `id_lang` int(11) unsigned NOT NULL,
  `name` varchar(255) NOT NULL,
  `address1` varchar(255) NOT NULL,
  `address2` varchar(255) DEFAULT NULL,
  `siret` varchar(18) DEFAULT NULL,
  `vat_number` varchar(28) DEFAULT NULL,
  `iban` varchar(28) DEFAULT NULL,
  `APE` varchar(15) DEFAULT NULL,
  `dni` varchar(15) DEFAULT NULL,
  `hours` text,
  `note` text,
  PRIMARY KEY (`id_store`, `id_lang`)
) ENGINE=ENGINE_TYPE DEFAULT CHARSET=utf8;

**Comment by Gruttersshop** [comment]
rest is simple after this.

**Comment by Gruttersshop** [comment]
testing update 
![image](https://user-images.githubusercontent.com/124369064/216725084-a96a8ed9-2798-4642-aa16-32ef93dd8b65.png)
